### PR TITLE
Fix incorrect GCS query

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
In cartographicToDbFromGcs(), the containsPoint() check was not enough to determine whether or not to try to use the GCS.

### CODE REVIEW

**Review by RBBentley — APPROVED** [review]

**Review by pmconne — CHANGES_REQUESTED** [review]
Let's clean up the noGcsDefined stuff while we're at it.

**Comment by markschlosseratbentley** [comment]
> Let's clean up the noGcsDefined stuff while we're at it.

@pmconne @bbastings That involves changing the project location extension, right? Since it currently calls 'disableGCS'.

**Comment by pmconne** [comment]
> @pmconne @bbastings That involves changing the project location extension, right? Since it currently calls 'disableGCS'.

We can keep disableGCS. @bbastings suggested changing that function to just set the GCS to `undefined`. I don't know if we'd want to set it back after the tool does its thing. If so, we could instead replace `_noGcsDefined` with `_gcsDisabled` and clean up all the stuff that wants to lazily set it to `false` after a failed request to the backend.

**Comment by bbastings** [comment]
@pmconne @markschlosseratbentley The project location extension can't be updated until there's a new published version, and it's backend (which is a 3rd repo for some reason is deployed with the new version). Not exactly sure how this all works...

I wasn't suggesting that disableGCS could set geographicCoordinateSystem to undefined, just that the extension *could* save and restore this value if disableGCS was removed. A _gcsDisabled flag would be fine.

Since the extension does call disableGCS with false to restore the previous state and also checks noGcsDefined directly, only removing _noGcsDefined and changing the implmentation of noGcsDefined and disableGCS would be feasible w/o requiring a change to the extension.

I do think removing the lazy setting of _noGcsDefined is a good thing...

**Comment by RBBentley** [comment]
> @pmconne @markschlosseratbentley The project location extension can't be updated until there's a new published version, and it's backend (which is a 3rd repo for some reason is deployed with the new version). Not exactly sure how this all works...
> 
> I wasn't suggesting that disableGCS could set geographicCoordinateSystem to undefined, just that the extension _could_ save and restore this value if disableGCS was removed. A _gcsDisabled flag would be fine.
> 
> Since the extension does call disableGCS with false to restore the previous state and also checks noGcsDefined directly, only removing _noGcsDefined and changing the implmentation of noGcsDefined and disableGCS would be feasible w/o requiring a change to the extension.
> 
> I do think removing the lazy setting of _noGcsDefined is a good thing...

Brien - how do you persist that the ECEF transform the user sets in your tool is used and that the GCS is ignored going forward? 
Do you clear the model location where the GCS definition is stored? -- If so, just making noGcsDefined non-lazy and initializing with geographicCoordinateSystem === undefined is sufficient right? -- I'm not sure that setting noGcsDefined to true on a GCS failure is necessary or correct - but users should test noGcsDefined before calling these methods (or catch the exception).

**Comment by bbastings** [comment]
@RBBentley The extension backend calls imodel.deleteFileProperty({ name: "DgnGCS", namespace: "dgn_Db" }); to remove the "unnamed" gcs when updating the ecef location. Not sure how this interacts w/Alain's stuff...hopefully the DgnGCS file property is still relevant...

**Comment by RBBentley** [comment]
> @RBBentley The extension backend calls imodel.deleteFileProperty({ name: "DgnGCS", namespace: "dgn_Db" }); to remove the "unnamed" gcs when updating the ecef location. Not sure how this interacts w/Alain's stuff...hopefully the DgnGCS file property is still relevant...

@AlainRobertAtBentley  -- Will removing the "gcs' property make geographicCoordinateSystem undefined when the iModel is reopened?

**Comment by markschlosseratbentley** [comment]
@pmconne @bbastings @RBBentley 
I just pushed changes to this branch/PR that rework the lazy-assignment of noGcsDefined.  Can I get a second set of eyes to check if it looks right?

File: core/frontend/src/IModelConnection.ts [hunk 1]
```diff
@@ -85,12 +85,12 @@ export abstract class IModelConnection extends IModel {
    * @internal
    */
   public readonly geoServices: GeoServices;
-  /** @internal Whether it has already been determined that this iModelConnection does not have a map projection. */
-  protected _noGcsDefined?: boolean;
-  /** @internal Whether it has already been determined that this iModelConnection does not have a map projection. */
-  public get noGcsDefined(): boolean | undefined { return this._noGcsDefined; }
+  /** @internal Whether GCS has been disabled for this iModelConnection. */
+  protected _gcsDisabled?: boolean;
```
Comment (bbastings) [inline core/frontend/src/IModelConnection.ts:89]: Should declare as true | undefined, since you don't check/allow false...

Comment (pmconne) [reply core/frontend/src/IModelConnection.ts:89]: Or just a boolean that defaults to false.

**Review by pmconne — APPROVED** [review]

File: core/frontend/src/IModelConnection.ts (hunk 1 above)
Comment (markschlosseratbentley) [reply core/frontend/src/IModelConnection.ts]: changed this.

Comment (markschlosseratbentley) [reply core/frontend/src/IModelConnection.ts]: @bbastings - resolve this comment if looks good?

Comment (bbastings) [reply core/frontend/src/IModelConnection.ts]: Why is noGCSDefined returning boolean | undefined? Would b cleaner just to make disableGCS a boolean that defaults to false and have noGCSDefined return true or false...

Comment (markschlosseratbentley) [reply core/frontend/src/IModelConnection.ts]: @bbastings Good point, thanks. The boolean | undefined was a holdover from the way the code used to be.  I just pushed a change to make it return boolean, as well as making _gcsDisabled a boolean defaulting to false.

**Pull request merged**